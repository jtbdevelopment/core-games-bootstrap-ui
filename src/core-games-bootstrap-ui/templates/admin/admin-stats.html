<div class="admin-stats">
    <div class="row">
        <table class="table table-striped table-condensed table-admin-states">
            <tr>
                <th></th>
                <th class="text-right">Last 24 hours</th>
                <th class="text-right">Last 7 days</th>
                <th class="text-right">Last 30 days</th>
            </tr>
            <tr class="text-right">
                <th class="text-left">Total Games</th>
                <td colspan="3">{{admin.gameCount}}</td>
            </tr>
            <tr class="text-right">
                <th class="text-left">Games Created</th>
                <td>{{admin.gamesLast24hours}}</td>
                <td>{{admin.gamesLast7days}}</td>
                <td>{{admin.gamesLast30days}}</td>
            </tr>
            <tr class="text-right">
                <th class="text-left">Total Players</th>
                <td colspan="3">{{admin.playerCount}}</td>
            </tr>
            <tr class="text-right">
                <th class="text-left">Players Created</th>
                <td>{{admin.playersCreated24hours}}</td>
                <td>{{admin.playersCreated7days}}</td>
                <td>{{admin.playersCreated30days}}</td>
            </tr>
            <tr class="text-right">
                <th class="text-left">Players Logged In</th>
                <td>{{admin.playersLastLogin24hours}}</td>
                <td>{{admin.playersLastLogin7days}}</td>
                <td>{{admin.playersLastLogin30days}}</td>
            </tr>
        </table>
    </div>
</div>
