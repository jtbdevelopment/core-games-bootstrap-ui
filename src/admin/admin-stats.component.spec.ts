import {fakeAsync, TestBed} from '@angular/core/testing';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {AdminStatsComponent} from './admin-stats.component';
import {BaseRequestOptions, ConnectionBackend, Http, RequestOptions} from '@angular/http';
import {MockBackend} from '@angular/http/testing';
import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
import {HttpRequest} from '@angular/common/http';

describe('admin stats component', () => {
    let httpMock: HttpTestingController;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbModule,
                HttpClientTestingModule
            ],
            declarations: [
                AdminStatsComponent
            ],
            providers: [
                {provide: ConnectionBackend, useClass: MockBackend},
                {provide: RequestOptions, useClass: BaseRequestOptions},
                Http
            ],
        });
        TestBed.compileComponents();
        httpMock = TestBed.get(HttpTestingController);
    });

    afterEach(() => {
        httpMock.verify();
    });

    it('should render results from http', fakeAsync(() => {
        let now: number = Math.floor((new Date()).getTime() / 1000);
        let approxTimes: number[] = [
            Math.floor((now - 86400) / 100),
            Math.floor((now - (86400 * 7)) / 100),
            Math.floor((now - (86400 * 30)) / 100)
        ];
        const fixture = TestBed.createComponent(AdminStatsComponent);
        fixture.detectChanges();

        let request = httpMock.expectOne((req: HttpRequest<any>) => req.url === '/api/player/admin/playerCount' &&
            req.responseType === 'text'
        );
        request.flush(1.2);

        request = httpMock.expectOne((req: HttpRequest<any>) => req.url === '/api/player/admin/gameCount' &&
            req.responseType === 'text'
        );
        request.flush(3.4);

        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersCreated/' + approxTimes[0]) &&
            req.responseType === 'text'
        );
        request.flush(2.3);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersCreated/' + approxTimes[1]) &&
            req.responseType === 'text'
        );
        request.flush(4.5);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersCreated/' + approxTimes[2]) &&
            req.responseType === 'text'
        );
        request.flush(5.6);

        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/gamesSince/' + approxTimes[0]) &&
            req.responseType === 'text'
        );
        request.flush(10.2);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/gamesSince/' + approxTimes[1]) &&
            req.responseType === 'text'
        );
        request.flush(10.3);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/gamesSince/' + approxTimes[2]) &&
            req.responseType === 'text'
        );
        request.flush(10.4);

        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersLoggedIn/' + approxTimes[0]) &&
            req.responseType === 'text'
        );
        request.flush(6.7);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersLoggedIn/' + approxTimes[1]) &&
            req.responseType === 'text'
        );
        request.flush(7.8);
        request = httpMock.expectOne((req: HttpRequest<any>) => req.url.startsWith('/api/player/admin/playersLoggedIn/' + approxTimes[2]) &&
            req.responseType === 'text'
        );
        request.flush(8.9);

        fixture.detectChanges();

        let content = fixture.nativeElement.querySelector('.admin-stats').textContent.trim();
        expect(content).toContain('1.2');
        expect(content).toContain('3.4');

        expect(content).toContain('2.3');
        expect(content).toContain('4.5');
        expect(content).toContain('5.6');

        expect(content).toContain('6.7');
        expect(content).toContain('7.8');
        expect(content).toContain('8.9');

        expect(content).toContain('10.2');
        expect(content).toContain('10.3');
        expect(content).toContain('10.4');
    }));
});
