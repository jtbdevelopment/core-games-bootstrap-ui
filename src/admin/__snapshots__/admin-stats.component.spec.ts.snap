// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin stats component should render results from http 1`] = `
<admin-stats
  gameCount={[Function Number]}
  gamesCreated={[Function Array]}
  http={[Function HttpClient]}
  playerCount={[Function Number]}
  playerLogins={[Function Array]}
  playersCreated={[Function Array]}
  time={[Function Number]}
  times={[Function Array]}
>
  <div
    class="admin-stats"
  >
    
    
    <table
      class="table table-striped table-sm table-admin-states"
    >
      
      
      <tbody>
        <tr>
          
        
          <th />
          
        
          <th
            class="text-right"
          >
            Last 24 hours
          </th>
          
        
          <th
            class="text-right"
          >
            Last 7 days
          </th>
          
        
          <th
            class="text-right"
          >
            Last 30 days
          </th>
          
        
          <th
            class="text-right"
          >
            All
          </th>
          
      
        </tr>
        
      
        <tr
          class="text-right"
        >
          
        
          <th
            class="text-left"
          >
            Games
          </th>
          
        
          <td>
            10.2
          </td>
          
        
          <td>
            10.3
          </td>
          
        
          <td>
            10.4
          </td>
          
        
          <td>
            3.4
          </td>
          
      
        </tr>
        
      
        <tr
          class="text-right"
        >
          
        
          <th
            class="text-left"
          >
            Players Created
          </th>
          
        
          <td>
            2.3
          </td>
          
        
          <td>
            4.5
          </td>
          
        
          <td>
            5.6
          </td>
          
        
          <td>
            1.2
          </td>
          
      
        </tr>
        
      
        <tr
          class="text-right"
        >
          
        
          <th
            class="text-left"
          >
            Players Logged In
          </th>
          
        
          <td>
            6.7
          </td>
          
        
          <td>
            7.8
          </td>
          
        
          <td>
            8.9
          </td>
          
        
          <td />
          
      
        </tr>
        
    
      </tbody>
    </table>
    
  
  </div>
  
</admin-stats>
`;
